core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -3002,6 +3002,32 @@ private void checkPushJoinThroughUnionOnRightDoesNotMatchSemiOrAntiJoin(JoinRelT
         .withProgram(program).check();
   }
 
+  @Test void testReduceConstantsProject() {
```

Comment:
how about name it more specifically, e.g. `testReducingConstantsInferedFromCorrelate`?

Reply:
Thanks for your suggestion, updated. And I rebase to update the commit message, please take a look again.